api/model/src/main/java/org/projectnessie/model/Reference.java

```diff
@@ -109,6 +120,27 @@ static String toPathString(String name, String hash) {
   @Schema(enumeration = {"branch", "tag"}) // Required to have lower-case values in OpenAPI
   enum ReferenceType {
     BRANCH,
-    TAG
+    TAG;
+
+    public static ReferenceType parse(String referenceType) {
+      try {
+        if (referenceType != null) {
+          referenceType = referenceType.toUpperCase(Locale.ROOT);
+          return "DETACHED".equals(referenceType) ? null : ReferenceType.valueOf(referenceType);
+        }
+        return null;
+      } catch (IllegalArgumentException e) {
+        return TAG;
```

Comment:
I'm trying to apply your ideas to #6646... Is the intent here to return a "random" enum constant when the text does not match any known enum constant?

Reply:
Not a random one, but the "likely best match" / least-common-denominator